### PR TITLE
Feature/bookingtest

### PR DESCRIPTION
**Pull request by RafSobolQAC**

### CODE REVIEW

File: test/services/ReleasedServicesTest.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import reactivemongo.play.json.collection.JSONCollection
+
+class ReleasedServicesTest extends PlaySpec with Results with MockitoSugar {
+  implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+
+  val mockedMongo: ReactiveMongoApi = mock[ReactiveMongoApi]
+  val mockFutureJSONColl: Future[JSONCollection] = mock[Future[JSONCollection]]
+  //      doReturn(mockFutureJSONColl).when(mockedMongo).database
+  var bookService: BookingServices = mock[BookingServices]
+  var releasedService: ReleasedServices = mock[ReleasedServices]
```
Comment (RafSobolQAC) [inline test/services/ReleasedServicesTest.scala]: Codacy Issue found: [Avoid mutable fields](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- test/controllers/BookingControllerTest.scala  3
- test/controllers/ReleasedControllerTest.scala  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100&bid=17280971)

File: test/services/ReleasedServicesTest.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import reactivemongo.bson.BSONObjectID
+import reactivemongo.play.json.collection.JSONCollection
+
+class ReleasedServicesTest extends PlaySpec with Results with MockitoSugar {
+  implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+
+  val mockedMongo: ReactiveMongoApi = mock[ReactiveMongoApi]
+  val mockFutureJSONColl: Future[JSONCollection] = mock[Future[JSONCollection]]
+  //      doReturn(mockFutureJSONColl).when(mockedMongo).database
+  var bookService: BookingServices = mock[BookingServices]
```
Comment (RafSobolQAC) [inline test/services/ReleasedServicesTest.scala]: Codacy Issue found: [Avoid mutable fields](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

File: test/services/BookingServicesTest.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+class BookingServicesTest extends PlaySpec with Results with MockitoSugar {
+
+  "The create-booking method" should {
+    "return a Future[WriteResult]" in {
+      val mockedMongo = mock[ReactiveMongoApi]
+      val mockFutureJSONColl = mock[Future[JSONCollection]]
+//      doReturn(mockFutureJSONColl).when(mockedMongo).database
+      var service = new BookingServices(mockedMongo)
+      var spyService = mock[BookingServices]
```
Comment (RafSobolQAC) [inline test/services/BookingServicesTest.scala]: Codacy Issue found: [Avoid mutable fields](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

File: app/controllers/BookingController.scala [hunk 4]
```diff
@@ -15,29 +15,63 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
 
-  def submitForm = Action.async {implicit request: Request[AnyContent] =>
-    Booking.createBookingForm.bindFromRequest.fold({formWithErrors =>
+  def submitSelectFilmForm(title: String) = Action.async { implicit request: Request[AnyContent] =>
+    titlesAndScreenings.map { titles =>
+
+      Ok(views.html.booking(Booking.createBookingForm,
+        title,
+        titles.filter(thing =>
+          thing._1 == title)
```
Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

File: app/controllers/BookingController.scala [hunk 5]
```diff
@@ -15,29 +15,63 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
   }
 
-  def submitForm = Action.async {implicit request: Request[AnyContent] =>
-    Booking.createBookingForm.bindFromRequest.fold({formWithErrors =>
+  def submitSelectFilmForm(title: String) = Action.async { implicit request: Request[AnyContent] =>
+    titlesAndScreenings.map { titles =>
+
+      Ok(views.html.booking(Booking.createBookingForm,
+        title,
+        titles.filter(thing =>
```
Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Usage of head on collections.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

File: app/controllers/BookingController.scala [hunk 6]
```diff
@@ -15,29 +15,63 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
-    Ok(views.html.booking(Booking.createBookingForm))
+  def showForm = Action.async { implicit request: Request[AnyContent] =>
+    titlesAndScreenings.map { titles =>
+      Ok(views.html.bookingfilm(Booking.getTitleForm, titles))
+    }
   }
 
-  def submitForm = Action.async {implicit request: Request[AnyContent] =>
-    Booking.createBookingForm.bindFromRequest.fold({formWithErrors =>
+  def submitSelectFilmForm(title: String) = Action.async { implicit request: Request[AnyContent] =>
```
Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [The method takes a value that is controlled by the client](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

File: test/services/BookingServicesTest.scala [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import reactivemongo.play.json.collection.JSONCollection
+
+class BookingServicesTest extends PlaySpec with Results with MockitoSugar {
+
+  "The create-booking method" should {
+    "return a Future[WriteResult]" in {
+      val mockedMongo = mock[ReactiveMongoApi]
+      val mockFutureJSONColl = mock[Future[JSONCollection]]
+//      doReturn(mockFutureJSONColl).when(mockedMongo).database
+      var service = new BookingServices(mockedMongo)
```
Comment (RafSobolQAC) [inline test/services/BookingServicesTest.scala]: Codacy Issue found: [Avoid mutable fields](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

File: test/controllers/ReleasedControllerTest.scala [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+          MovieWithID(BSONObjectID.generate(), "image2", "title3", "director", "descaaa", List("act3", "act2"), List("morning", "evening"))
+        )
+      }
+      when(releasedService.getMovies).thenReturn(filmList)
+      val result: Future[Result] = controller.getOnlyMoviesAndScreenings.apply(FakeRequest())
+      contentType(result) mustBe Some("text/plain")
+      filmList.map{
+        value =>
+          {
+            contentAsString(result) must not contain value.head.description
```
Comment (RafSobolQAC) [inline test/controllers/ReleasedControllerTest.scala]: Codacy Issue found: [Usage of head on collections.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Clones added
============
- test/controllers/ReleasedControllerTest.scala  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100&bid=17280971)

File: test/controllers/ReleasedControllerTest.scala [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+          MovieWithID(BSONObjectID.generate(), "image2", "title3", "director", "descaaa", List("act3", "act2"), List("morning", "evening"))
+        )
+      }
+      when(releasedService.getMovies).thenReturn(filmList)
+      val result: Future[Result] = controller.getOnlyMoviesAndScreenings.apply(FakeRequest())
+      contentType(result) mustBe Some("text/plain")
+      filmList.map{
+        value =>
+          {
+            if (value.nonEmpty) contentAsString(result) must not contain value.head.description
```
Comment (RafSobolQAC) [inline test/controllers/ReleasedControllerTest.scala]: Codacy Issue found: [Usage of head on collections.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

File: app/controllers/BookingController.scala [hunk 10]
```diff
@@ -15,29 +15,61 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
-    Booking.createBookingForm.bindFromRequest.fold({formWithErrors =>
+
+  def submitSelectFilmFormSubmit = Action.async { implicit request: Request[AnyContent] =>
+    Booking.getTitleForm.bindFromRequest.fold({ formWithErrors =>
+      Future.successful(BadRequest(views.html.bookingfilm(formWithErrors, List())))
+    }, { film =>
+      titlesAndScreenings.map { films =>
+        Ok(views.html.booking(Booking.createBookingForm,
+          film,
+          films.filter(movie =>
```
Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Usage of head on collections.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

File: app/controllers/BookingController.scala [hunk 11]
```diff
@@ -15,29 +15,61 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
+            movie._1 == film)
```
Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones added
============
- test/controllers/ReleasedControllerTest.scala  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100&bid=17280971)

File: app/controllers/BookingController.scala [hunk 12]
```diff
@@ -15,29 +15,60 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
+
+  def submitSelectFilmFormSubmit = Action.async { implicit request: Request[AnyContent] =>
+    Booking.getTitleForm.bindFromRequest.fold({ formWithErrors =>
+      Future.successful(BadRequest(views.html.bookingfilm(formWithErrors, List())))
+    }, { film =>
+      titlesAndScreenings.map { films =>
+        Ok(views.html.booking(Booking.createBookingForm,
+          film,
+          films.find(movie =>
+            movie._1 == film).getOrElse(("None", List()))._2))
```
Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

File: app/controllers/BookingController.scala [hunk 13]
```diff
@@ -15,29 +15,60 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
-    Booking.createBookingForm.bindFromRequest.fold({formWithErrors =>
+
+  def submitSelectFilmFormSubmit = Action.async { implicit request: Request[AnyContent] =>
+    Booking.getTitleForm.bindFromRequest.fold({ formWithErrors =>
+      Future.successful(BadRequest(views.html.bookingfilm(formWithErrors, List())))
+    }, { film =>
+      titlesAndScreenings.map { films =>
+        Ok(views.html.booking(Booking.createBookingForm,
+          film,
+          films.find(movie =>
```
Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- test/controllers/ReleasedControllerTest.scala  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100&bid=17280971)

File: app/controllers/BookingController.scala [hunk 14]
```diff
@@ -15,29 +15,64 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
+    Booking.getTitleForm.bindFromRequest.fold({ formWithErrors =>
+      Future.successful(BadRequest(views.html.bookingfilm(formWithErrors, List())))
+    }, { film =>
+      titlesAndScreenings.map { films =>
+
+        Ok(views.html.booking(Booking.createBookingForm,
+          film,
+//          films.find(movie =>
+//            movie._1 == film).getOrElse(("None", List()))._2))
+          films.find { case ((titleOfFilm, _))  =>
```
Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- test/controllers/ReleasedControllerTest.scala  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100&bid=17280971)

File: app/controllers/BookingController.scala [hunk 15]
```diff
@@ -15,29 +15,64 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
+    Booking.getTitleForm.bindFromRequest.fold({ formWithErrors =>
+      Future.successful(BadRequest(views.html.bookingfilm(formWithErrors, List())))
+    }, { film =>
+      titlesAndScreenings.map { films =>
+
+        Ok(views.html.booking(Booking.createBookingForm,
+          film,
+//          films.find(movie =>
+//            movie._1 == film).getOrElse(("None", List()))._2))
+          films.find { case ((titleOfFilm, screenings))  =>
```
Comment (RafSobolQAC) [inline app/controllers/BookingController.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- test/controllers/ReleasedControllerTest.scala  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100&bid=17280971)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- test/controllers/ReleasedControllerTest.scala  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100&bid=17280971)

File: app/models/MovieWithID.scala [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+case class Movie(
+                  var image: String,
+                  var title: String,
+                  var director: String,
+                  var description: String,
+                  var actors: List[String],
+                  var screenings: List[String])
+
+case class MovieWithID(
+                        _id: BSONObjectID,
```
Comment (RafSobolQAC) [inline app/models/MovieWithID.scala]: Codacy Issue found: [Field name '_id' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277100)

**Review by johnpooler — APPROVED** [review]
nice tests

**Pull request merged**